Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionTabCocoa.mm

```diff
@@ -627,6 +627,14 @@
 
 void WebExtensionTab::captureVisibleWebpage(CompletionHandler<void(CocoaImage *, Error)>&& completionHandler)
 {
+    bool canUseCaptureVisibleWebpage = isValid() && m_respondsToCaptureVisibleWebpage;
+    __weak WKWebView *mainWebView = !canUseCaptureVisibleWebpage ? this->mainWebView() : nil;
+
+    if (!canUseCaptureVisibleWebpage && !mainWebView) {
+        completionHandler(nil, toErrorString(@"tabs.captureVisibleTab()", nil, @"capture is unavailable for this tab"));
+        return;
+    }
```

Comment:
I think this would read clearer as `delegateIsUnavailable`, to avoid `!` everywhere. I also don't think `__weak` is needed here, is it?

```
bool delegateIsUnavailable = !isValid() || !m_respondsToCaptureVisibleWebpage;
WKWebView *mainWebView = delegateIsUnavailable ? this->mainWebView() : nil;

if (delegateIsUnavailable && !mainWebView) {
    completionHandler(nil, toErrorString(@"tabs.captureVisibleTab()", nil, @"capture is unavailable for this tab"));
    return;
}
```

Reply:
Done!